core/src/swarm/registry.rs

```diff
@@ -53,7 +56,8 @@ impl Addresses {
     pub fn new(limit: NonZeroUsize) -> Self {
         Addresses {
             limit,
-            registry: SmallVec::new()
+            registry: SmallVec::new(),
+            reports: VecDeque::with_capacity(200),
```

Comment:
Note that 200 is only a lower bound and I guess `VecDeque` will allocate the nearest power of 2 bytes.